yt_dlp/extractor/safari.py

```diff
@@ -193,7 +193,11 @@ def _real_extract(self, url):
         part = self._download_json(
             url, '%s/%s' % (mobj.group('course_id'), mobj.group('part')),
             'Downloading part JSON')
-        return self.url_result(part['web_url'], SafariIE.ie_key())
+        webUrl = part['web_url']
+        webUrl = webUrl.replace('library/view', 'videos')
+        naturalKeys = part['natural_key']
+        webUrl = webUrl[:webUrl.rfind('/') + 1] + naturalKeys[0] + '-' + naturalKeys[1][:-5]
```

Comment:
If O'Reilly fixes the video URL at their API output this string manipulation will not work. I believe it would be better to replace `library/view` with videos and do the rest of the manipulation if only there is a match.